docker/scripts/New-AzureStackTestResources.ps1

```diff
@@ -0,0 +1,127 @@
+# Run Login-Environment.ps1 before running this script.
+
+[CmdletBinding()]
+param (
+    [string]
+    $Location = $ENV:Location
+)
+
+$resourceGroup = "azurestack-container-rg"
+$offer = "azurestack-container-offer"
+$plan = "azurestack-container-plan"
+
+function New-Quotas
+{
+    $serviceQuotas  = @()
+
+    $quotaName                  = "azstorquota"
+    $capacityInGb               = 1000
+    $numberOfStorageAccounts    = 200
+    $storageQuota = New-AzsStorageQuota -Name $quotaName -CapacityInGb $capacityInGb -NumberOfStorageAccount $numberOfStorageAccounts
+    $serviceQuotas += $storageQuota.Id
+
+    $quotaName      = "azcompquota"
+    $vmCount        = 100
+    $coresLimit     = 100
+    $availabilitySetCount = 10
+    $vmScaleSetCount = 100
+    $computeQuota = New-AzsComputeQuota -Name $quotaName `
+        -AvailabilitySetCount $availabilitySetCount `
+        -CoresLimit $coresLimit `
+        -VmScaleSetCount $vmScaleSetCount `
+        -VirtualMachineCount $vmCount `
+        -PremiumManagedDiskAndSnapshotSize 1000 `
+        -StandardManagedDiskAndSnapshotSize 1000
+    $serviceQuotas += $computeQuota.Id
+
+    $quotaName                      = "aznetwquota"
+    $publicIpsPerSubscription       = 50
+    $vNetsPerSubscription           = 50
+    $gatewaysPerSubscription        = 1
+    $connectionsPerSubscription     = 2
+    $loadBalancersPerSubscription   = 50
+    $nicsPerSubscription            = 100
+    $securityGroupsPerSubscription  = 50
+    $networkQuota = New-AzsNetworkQuota -Name $quotaName `
+        -MaxNicsPerSubscription $nicsPerSubscription `
+        -MaxPublicIpsPerSubscription $publicIpsPerSubscription `
+        -MaxVirtualNetworkGatewayConnectionsPerSubscription $connectionsPerSubscription `
+        -MaxVnetsPerSubscription $vNetsPerSubscription `
+        -MaxVirtualNetworkGatewaysPerSubscription $gatewaysPerSubscription `
+        -MaxSecurityGroupsPerSubscription $securityGroupsPerSubscription `
+        -MaxLoadBalancersPerSubscription $loadBalancersPerSubscription
+    $serviceQuotas += $networkQuota.Id
+
+    $keyvaultQuota = Get-AzsKeyVaultQuota -Location $Location
+    $serviceQuotas += $keyvaultQuota.Id
+
+    $serviceQuotas
+}
+
+function Set-Offer
+{
+    $getResourceGroup = Get-AzResourceGroup -Name $resourceGroup
+    if ($null -eq $getResourceGroup)
+    {
+        New-AzResourceGroup -Name $resourceGroup -Location $Location
+    }
+
+    $getPlan = Get-AzsPlan -ResourceGroupName $resourceGroup -Name $plan
+    if ($null -eq $getPlan)
+    {
+        $serviceQuotas = New-Quotas
+        $getPlan = New-AzsPlan -Name $plan `
+            -DisplayName $plan `
+            -Location $Location `
+            -ResourceGroupName $resourceGroup `
+            -QuotaIds $serviceQuotas `
+            -ErrorAction Stop
```

Comment:
pls set the ErrorActionPreference as Stop at the start of the script rather than using it per cmdlet